geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/BaseCommandQuery.java

```diff
@@ -177,20 +169,15 @@ protected boolean processQueryUsingParams(final Message msg,
 
         // Get the collection type (which includes the element type)
         // (used to generate the appropriate instance on the client)
-        collectionType = getCollectionType(selectResults);
-        isStructs = collectionType.getElementType().isStructType();
+        CollectionType collectionType = getCollectionType(selectResults);
+        boolean isStructs = collectionType.getElementType().isStructType();
 
         // Check if the Query is from CQ execution.
         if (cqQuery != null) {
-          // Check if the key can be sent to the client based on its version.
-          sendCqResultsWithKey = sendCqResultsWithKey(servConn);
-
-          if (sendCqResultsWithKey) {
-            // Update the collection type to include key info.
-            collectionType = new CollectionTypeImpl(Collection.class,
-                new StructTypeImpl(new String[] {"key", "value"}));
-            isStructs = collectionType.getElementType().isStructType();
-          }
+          // Update the collection type to include key info.
```

Comment:
presume this means all existing/supported client versions can handle the key...

Reply:
Yes, the corresponding method `sendCqResultsWithKey()` would now always return true. I refactored it out based on static analyzer.